### ISSUE TITLE
Grouped text is not placed correctly

### ISSUE THREAD

**Issue by mmakrzem** [svgdotjs/svg.js#944]
# Bug report

> **For support questions, please use [stackoverflow](https://stackoverflow.com/questions/tagged/svg.js) with the tag svg.js or head to our chat over at [gitter](https://gitter.im/svgdotjs/svg.js)**.

## Fiddle

https://jsfiddle.net/mmakrzem/3gc0ny2k/

## Explanation

I placed some text located at 50,50 (green in color). ok

If I place the same text in a group, and transform the group to be at 50,50, the text does not render at the same spot. See red text vs green text in the fiddle

**Comment by Fuzzyma** [comment]
This was fixed in 3.0.5. The new behavor of group movement should place the text exactly where you want it to be. Can you try it and confirm, that it works?
Since there are no docs for 3.0 yet, feel free to head over to gitter for better support.

**Comment by mmakrzem** [comment]
@Fuzzyma , how do I get v3.0.5?  On npm I only see v2.7.1 https://www.npmjs.com/package/svg.js

**Comment by Fuzzyma** [comment]
We created an Organisation on npm.js. That's why our package is now scoped. You have to run `npm install @svgdotjs/svg.js`. That should give you the current version.

**Comment by Fuzzyma** [comment]
To enable treeshaking, main.js only exports all submodules as named exports. Basically everything which was accessible through `SVG.*` is exported as submodule. The SVG function itself is available as named export too.

So you can import stuff like this:

```
import { SVG, Color, Rect } from '@svgdotjs/svg.js'
```

If you don't care about treeshaking and just wand your old behavior back, you can explicitly import `'svg.js-folder/src/svg.js'`. This gives you the old SVG object as default export

Hope that helps!

**Comment by mmakrzem** [comment]
@Fuzzyma I can't import SVG, I get this error:

[ ERROR ]  TypeScript: src/app/MyApp.ts:1:9
           Module '"C:/Users/me/node_modules/@svgdotjs/svg.js/svg.js"' has no
           exported member 'SVG'.

      L1:  import { SVG } from '@svgdotjs/svg.js';

**Comment by mmakrzem** [comment]
[ ERROR ]  TypeScript: src/app/MyApp.ts:2:20
           Could not find a declaration file for module '@svgdotjs/svg.js/src/svg.js'.
           'C:/Users/me/node_modules/@svgdotjs/svg.js/src/svg.js' implicitly has an
           'any' type. Try `npm install @types/svgdotjs__svg.js` if it exists or add a new declaration (.d.ts) file
           containing `declare module '@svgdotjs/svg.js/src/svg.js';`

      L1:  //import { SVG } from '@svgdotjs/svg.js';
      L2:  import { SVG } from '@svgdotjs/svg.js/src/svg.js';

**Comment by mmakrzem** [comment]
Also, the documentation found here: https://svgjs.com/docs/3.0/installation/ needs to be updated

**Comment by Fuzzyma** [comment]
I am not familiar with typescript and therefore don't know how to handle that errors. It seems like the library was imported just fine. It just don't know what was imported.

For the docs: yes they needs an update. We are working on it. Feel free to create a PR and fix what you know.

**Comment by mmakrzem** [comment]
@Fuzzyma okay, lets go back to using `import { SVG } from '@svgdotjs/svg.js';` I see there is a typescript definition file (svg.js.d.ts) included in the project.  How is this file being created?  It doesn't look any different then the file that was included in the old project: https://github.com/svgdotjs/svg.js/blob/master/svg.js.d.ts

**Comment by Fuzzyma** [comment]
Lets meet on gitter and talk there. It seems to be more complicated that I thought it to be. We can post the result here later: https://gitter.im/svgdotjs/svg.js

**Comment by drparadoxical** [comment]
Heads up, you're both fighting a losing battle here. SVG 1.1, relative to applying matrix transformations to multi-line text elements (both with and without tspan elements buried within), drops off here, and layout engines haphazardly implemented SVG text interpretation over the years. 

Your hard ceiling is essentially this: try to create a single text element, at least two lines, center or right aligned, with an arc applied evenly and correctly to the text element itself. Go ahead, try it. 

Also, @Fuzzyma - I've been using this library for years now, and see you guys are trying to push hard on meeting the W3C SVG 2 recommendation upon its release. We're doing the same. I've had to override significant portions in order to bring complex text transformations (and hell, just consistently positioning elements cross-browser, honestly...) into the fold. Want some help? I'd love to join the fray. 

@mmakrzem - I've suffered enough for all of us. Let me get a patch or PR in, or get some snippets for you, to save you a great deal of time and sanity. 

SVG 1.1 is a harsh mistress, but oh how beautiful she can be when she wants to be.

**Comment by drparadoxical** [comment]
@mmakrzem - I believe this issue boils down to nothing more than commands being out of order. Take a look at this fiddle, and see how I create the group before adding the single and multiple lined text elements? https://jsfiddle.net/drparadox/e2aq8fdn/12/

What I mentioned in my previous comment doesn't relate to what's going on here, but I'm still going to push in a PR so svg.js can handle ridiculousness such as arced, skewed, rotated, multiline, varied size and line-height, insanity text blocks.

**Comment by mmakrzem** [comment]
@drparadoxical you are correct.  If I leave the code as is here https://jsfiddle.net/mmakrzem/5jh2tp6e/1/ then there is a rendering problem.  However if I uncomment the move(0,0) line then the problem fixes itself.

**Comment by drparadoxical** [comment]
Good, then we're both crazy. I'm fine with that.

**Comment by Fuzzyma** [comment]
**TL;DR** There is an easy solution to the usecase in that issue. However, I didnt know that it was critical to fix it and took this issue as bug report only - sorry**

Wow this issue eploded!
@drparadoxical sure go ahead and jump on the train. But before you start coding, create an issue eplaining exactly which problem you want to solve with the PR. I am hones, I am not sure about what you are talking.

For the magic in @mmakrzem example let me explain it.

There are 2 things you need to know:

**Groups dont have a geometry themselves**. A group is just a grouping of elements which then can be transformed together. So its always as big as its contents.
There is no difference in moving the text alone or while in a group. It renders the same results:

```js
canvas.text('foo').move(0,0)
// same as
canvas.group().text('foo').move(0,0)
```

The second thing is: **Text is positioned by its baseline by default**. The baseline us the line you are basically writing on. It is NOT the upper left edge.
However, ALL other shapes in svg.js can be moved by its upper left edge. So we decided, that `move` on a text would do the same. So when you call `move(0,0)` it tries to position the upper left edge at 0, 0.
However, when you dont call it, you dont set any x or y so it defaults to 0 which means the baseline is positioned at 0, 0 instead.

So what you did by leaving out the 0, 0 was actually the first case I showed but without the move at all.
You never positioned your text at its upper left edge in your group example so it is still positioned at its baseline beeing 0, 0

So to fix your example you just move your text to 50, 50 in both cases.
If you dont want that in the group example, at least move the text to 0, 0 before translating the group:

https://jsfiddle.net/mmakrzem/3gc0ny2k/

It is really that easy. However, I wasnt aware, that you didnt find a workaround for this. I thought that is a "just" a bug report. Because it is indeed stupid to have to call `move(0,0)` all the time. In 3.0 there is a method `amove` on text which moves the text by its baseline. So `canvas.text(0,0).amove(50, 50)` will actually place the text by its baseline. Using that would remove that burden if calling move(0,0) on every text creation

Hope that wall of text helps
